examples/CustomCollectionView-Swift/Sample/ViewController.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  ViewController.swift
+//  Sample
+//
+//  Created by Rajeev Gupta on 11/9/16.
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+//  FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+//  ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+//  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import UIKit
+import AsyncDisplayKit
+
+class ViewController: UIViewController, MosaicCollectionViewLayoutDelegate, ASCollectionDataSource, ASCollectionDelegate {
+  
+  var _sections: [[UIImage]]?
+  var _collectionNode: ASCollectionNode?
+  var _layoutInspector: MosaicCollectionViewLayoutInspector?
```

Comment:
None of these need to be optionals. Initialize sections and layoutInspector here and the collectionNode before `super.init`.
The collectionNode and the layoutInspector will never change, so convert those two to `let` statements.